### DISCUSSION TITLE
Emoji Support

### DISCUSSION THREAD

**Discussion by unrealapex** [kyx0r/nextvi#139] [Q&A] [answered]
Emoji rendering would be nice when viewing files with emojis and other characters in them. Currently Nextvi renders only a small subset of emojis. Below is a test file that contains every emoji. When opening the file in Nextvi, most of the characters are rendered as the Unicode replacement character(`�`):
[all_emojis.txt](https://github.com/user-attachments/files/17689648/all_emojis.txt)

**Additional info**
My terminal supports unicode and I have Fontconfig setup correctly.

**Comment by kyx0r** [comment, marked as answer]
No idea, I have never used multi codepoint emojis in the terminal.

See if they render if you run
```:uc```

**Reply by unrealapex** [reply]
Would you recommend I keep this in my exrc or only use it when I need it?

**Reply by kyx0r** [reply]
Well, :uc is not really a solution as it will mess up cursor position on everything else. I think there is an array called bchars in uc.c which checks for intermediate codepoints. Maybe if you remove it things will work. I'm not too sure myself weather this list is still needed or not tbh. Inherited from neatvi

**Reply by kyx0r** [reply]
> Would you recommend I keep this in my exrc or only use it when I need it?

After careful examination, I've determined to add official emoji support, which can be enabled using these 2 new toggle commands:
:uz:ub

:uz renders zero width chars invisible and :ub disables the bchars placeholders.

This should make it flexible enough to render all kinds of data by leveraging these toggles.

Enjoy!
